code/controllers/configuration/entries/jobs.dm

```diff
@@ -0,0 +1,184 @@
+// This file pretty much just handles all of the interactions between jobconfig.toml and the codebase. This is started by work originating in SSconfig, so I'm okay with it being here.
+
+/// Initializes all of the config singletons for each job config type and adds it to the `job_config_datum_singletons` var list.
+/datum/controller/subsystem/job/proc/generate_config_singletons()
+	for(var/datum/job_config_type/config_datum as anything in subtypesof(/datum/job_config_type))
+		var/config_datum_key = initial(config_datum.name)
+		job_config_datum_configurables += config_datum_key
+		var/new_config_datum = new config_datum
+		job_config_datum_singletons[config_datum_key] = new_config_datum
+
+/// Called in jobs subsystem initialize if LOAD_JOBS_FROM_TXT config flag is set: reads jobconfig.toml (or if in legacy mode, jobs.txt) to set all of the datum's values to what the server operator wants.
+/datum/controller/subsystem/job/proc/load_jobs_from_config()
+	var/toml_file = "[global.config.directory]/jobconfig.toml"
+
+	if(!legacy_mode) // this flag is set during the setup of SSconfig, and all warnings were handled there.
+		var/job_config = rustg_read_toml_file(toml_file)
+
+		for(var/datum/job/occupation as anything in joinable_occupations)
+			var/job_title = occupation.title
+			var/job_key = occupation.config_tag
+			if(!job_config[job_key]) // Job isn't listed, skip it.
+				// List both job_title and job_key in case they de-sync over time.
+				message_admins(span_notice("[job_title] (with config key [job_key]) is missing from jobconfig.toml! Using codebase defaults."))
+				continue
```

Comment:
bit weird to pull out job_title into its own vars despite it only being used once. I realize this isn't today you, but do you mind removing that?